# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [2]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [6]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose (row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
late_payment = data[(data['debt'] == 1)].count() / data[(data['debt'] == 0)].count()
late_payment['debt']*100
#Просрочка в % по кридиту общая.

8.837185570692382

Получившиеся значение в 8.8% должников репрезентативна и может помочь для дальнейших выводов.

In [23]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
debt_and_children = data.pivot_table(index=['debt'], columns='children', values = 'family_status_id', aggfunc='count')

#Предобработка данных - 3.Изменение типов данных - 8.Сводные таблицы
#index — столбец или столбцы, по которым группируют данные
#columns — столбец, по значениям которого происходит группировка
#values — значения, по которым мы хотим увидеть сводную таблицу
#aggfunc — функция, применяемая к значениям

ch0 = debt_and_children[0][1] / debt_and_children[0][0]
ch1 = debt_and_children[1][1] / debt_and_children[1][0]
ch2 = debt_and_children[2][1] / debt_and_children[2][0]
ch3 = debt_and_children[3][1] / debt_and_children[3][0]
ch4 = debt_and_children[4][1] / debt_and_children[4][0]
print("{0:.2f}% Нет детей".format(ch0*100))
print("{0:.2f}% 1 Ребенок".format(ch1*100))
print("{0:.2f}% 2 Ребенка".format(ch2*100))
print("{0:.2f}% 3 Ребенка".format(ch3*100))
print("{0:.2f}% 4 Ребенка".format(ch4*100))

8.16% Нет детей
10.17% 1 Ребенок
10.44% 2 Ребенка
8.91% 3 Ребенка
10.81% 4 Ребенка


In [37]:
chil_debt = data.groupby('children').agg({'debt':['count','sum']})
chil_quality = chil_debt['debt']['sum']/chil_debt['debt']['count']
chil_quality*100 

children
0    7.543822
1    9.234609
2    9.454191
3    8.181818
4    9.756098
5    0.000000
dtype: float64

In [25]:
# это твой код с соответствующим output'ом

debt_and_children = data.pivot_table(index=['debt'], columns='children', values = 'family_status_id', aggfunc='count')

debt_and_children

children        0       1       2      3     4    5
debt                                               
0         13028.0  4364.0  1858.0  303.0  37.0  9.0
1          1063.0   444.0   194.0   27.0   4.0  NaN

In [26]:
# 1.Мне здесь не нравиться, что в индексе не количество детей, а в столбцах не признак долга (поэтому я сейчас эти 2 параметра 
# "махну местами"). Резульат будет привычно вытянут вниз, вместо непривычного - "по горизонтали" (хотя это дело вкуса).

debt_and_children = data \
                        .pivot_table(index='children', 
                                     columns='debt', 
                                     values = 'family_status_id', 
                                     aggfunc='count',
                                     fill_value=0)

debt_and_children

debt          0     1
children             
0         13028  1063
1          4364   444
2          1858   194
3           303    27
4            37     4
5             9     0

In [27]:
# 2.Ну и нахождение % должников у меня короче, в одну строку.

debt_and_children['%'] = round((debt_and_children[1] / (debt_and_children[0] + debt_and_children[1])) * 100, 2)

debt_and_children

debt          0     1     %
children                   
0         13028  1063  7.54
1          4364   444  9.23
2          1858   194  9.45
3           303    27  8.18
4            37     4  9.76
5             9     0  0.00

**Вывод:** 
Люди с детьми имеют задолжность чаще чем без детей. Причем из за вклада в общую просрочку оплаты, долг вырастает больше чем на пол процента. Разница довольно существенная между нет детей и есть дети, можно было бы пообщатся со специалистами банка. Обратить внимание на это провителство и попытатся договарится о субсидиях людям с детьми как наиболее приорететным и незащещенным слоям населения. К примеру льготная ипотека повышает цены на квартиры из за увеличения спроса на них, но расходует деньги за счет покрытия издержек. Кредиты это топлива роста, но и здесь надо понимать риски и сохранять баланс. Работать на опережения. Статистика прекрасный инструмент, но и ее надо применять с умом.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [28]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
family_grouped=data.groupby('family_status').agg({'debt':['count','sum']}) 
family_debt_quality=family_grouped['debt']['sum']/family_grouped['debt']['count']
family_debt_quality*100

family_status
Не женат / не замужем    9.763948
в разводе                7.064760
вдовец / вдова           6.624606
гражданский брак         9.313014
женат / замужем          7.560558
dtype: float64

**Вывод:** Тут тоже есть существенная кореляция. К примеру видно что вдовец / вдова почти на треть чаще вовремя платят чем Не женат / не замужем . Хотя тут могут сочетатся такие факторы как возрост. к примеру можно предположить, что большенство тех кто развелся или потерял супруга/супругу старше холостых или начавших свои отношения. Эту теорию можно проверить, но данных кто когда овдовел нет так что выводы будут косвенными.

In [32]:
dob_years=data.groupby('dob_years').agg({'debt':['count','sum']}) 
dob_years_quality=dob_years['debt']['sum']/dob_years['debt']['count']
dob_years_quality*100

dob_years
0      8.000000
19     7.142857
20     7.843137
21    12.727273
22    13.661202
23     8.000000
24     9.125475
25    12.078652
26    11.576355
27    10.000000
28    11.377246
29    11.992620
30    10.131332
31    13.693694
32    10.099010
33     9.532062
34    10.588235
35     7.667210
36     7.789855
37     9.622642
38    10.774411
39     7.705779
40     7.820300
41     8.153078
42     7.106599
43     8.644401
44     7.578558
45     7.910751
46     8.154506
47     7.756813
48     6.179775
49     6.138614
50     9.055118
51     5.405405
52     5.797101
53     7.658643
54     6.553911
55     5.215420
56     6.694561
57     6.622517
58     5.286344
59     6.818182
60     5.898123
61     2.840909
62     5.187320
63     4.477612
64     4.651163
65     7.253886
66     4.395604
67     5.389222
68     8.080808
69     4.819277
70     4.615385
71     1.785714
72     6.060606
73     0.000000
74     0.000000
75     0.000000
dtype: float64

**Вывод:** Как видно из таблицы люди до 35 чаще невыплачивают задолжность. Хоть есть года с довольно высоким показателем как и с аномально низким, но это может быть из за относительно неравномерного распределения полученных данных либо активности людей.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [30]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
total = data.pivot_table(index=['debt'], columns='total_income_category', values='total_income', aggfunc='count')

in1 = total['E'][1] / total['E'][0]
in2 = total['D'][1] / total['D'][0]
in3 = total['C'][1] / total['C'][0]
in4 = total['B'][1] / total['B'][0]
in5 = total['A'][1] / total['A'][0]

print("{0:.2f}% Категория E (Доход от 0 до 30000 р.)".format(in1*100))
print("{0:.2f}% Категория D (Доход от 30001 до 50000 р.)".format(in2*100))
print("{0:.2f}% Категория C (Доход от 50001 до 200000 р.)".format(in3*100))
print("{0:.2f}% Категория B (Доход от 200001 до 1000000 р.)".format(in4*100))
print("{0:.2f}% Категория A (Более 1000000 р.)".format(in5*100))

10.00% Категория E (Доход от 0 до 30000 р.)
6.40% Категория D (Доход от 30001 до 50000 р.)
9.29% Категория C (Доход от 50001 до 200000 р.)
7.60% Категория B (Доход от 200001 до 1000000 р.)
8.70% Категория A (Более 1000000 р.)


**Вывод:** Разница имеется. Чаще отдают люди с доходом от 30000 до 50000 рублей. Если категорию Е (0-30000) можно обяснить нехваткой денег то почему люди с доходом выше 50000 платят реже? Я предпологаю, что это из за других условий кредита, и сумм кредитования которые резко отличаются для них. Но это надо уточнять у специалистов банка. 

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [31]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
purpose = data.pivot_table(index=['debt'], columns='purpose_category', values='total_income_category', aggfunc='count')

in1 = purpose['операции с автомобилем'][1] / purpose['операции с автомобилем'][0]
in2 = purpose['операции с недвижимостью'][1] / purpose['операции с недвижимостью'][0]
in3 = purpose['получение образования'][1] / purpose['получение образования'][0]
in4 = purpose['проведение свадьбы'][1] / purpose['проведение свадьбы'][0]


print("{0:.2f}% операции с автомобилем".format(in1*100))
print("{0:.2f}% операции с недвижимостью".format(in2*100))
print("{0:.2f}% получение образования".format(in3*100))
print("{0:.2f}% проведение свадьбы".format(in4*100))

10.31% операции с автомобилем
7.82% операции с недвижимостью
10.20% получение образования
8.59% проведение свадьбы


**Вывод:** В различных категориях, различный процент задолжности. Такая кореляция скорее всего из за того, что при получении квартиры часто она служит горантом уплаты. Образование чаще требуется молодым людям, а они чаще перестают платить (как видно из таблице dob_years), довольно сложно учится и работать одновременно.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Причины могут быть различными в зависимости от ситуации. По большей части пропуски зависят от системы которая собирает данные. Если при заходе на сайт он фиксирует время прибывания и сам дает уникальный номер посетившего, имя  и семейное положение может получть только при регистрации посетителя на сайте, к примеру уровень доходов можно указать по желанию, то и уровень заполнености будет разный. Да и сама информация заполненная в ручную посетителем может быть неверной, поэтому анамалии при исследовании тоже оень важны. В исследовании даже пропуски могут быть важной информацией.
В таблице предложенной нам пропуски были в total_income (суммарный доход) и days employed (дни работы). Могу предположить, что эти люди неимеют постоянного легального дохода. Они живут либо за счет мецинатов (родственников, содержантов), либо получают деньги в конверти (имеют устную договаренность и их доход не фиксируется). Могу предположить, что сам практикум удалил данные для дополнительных заданий и показа важности медианы, но против этой теории говорит одно итоже количества пропусков в 2 колонках, при этом данные days employed неиспользуются. Возможно days employed использовалась ранее или используется в других практикумах.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:*  Поскольку очень высокие зарплаты как и очень низкие могут сильно сдвинуть среднее значения и непоказать картины для большенства. Из за этого могут быть приняты неверные решения. Также и медиану лучше брать для каждого типа заначений отдельно, это тоже поможет увидить картину более точно.

### Шаг 4: общий вывод.

Анализируя по разным групам я подумал, что возраст тоже может являтся одним из ключевых факторов и проверил свою гипотизу. Она показала определенную тенденцию, но поней нельзя сделать однозначного вывода. Тут и косвенные выводы показывают разные значения. К примеру чем больше детей тем скорее всего старше человек, но у кого нет детей тот более надежный платильщик (это логично поскольку меньше обязательств и расходов), но в целом обычно саршие люди имеют более стабильный доход. Как и везде существует множество факторов и противовесов, жизненных условий и обстоятельств меняющих те или иные показатели. Конечно для данной сферы среднее 8.8% должников и разница в +-2% являются существенными, но для всестороннего анализа и выробатки мер по улучшению работы и уменьшению задолжности нужно проводить более всесторонний анализ и работать с професионалами разных сфер. Но даже такие иследования могут показать в каком направлении стоит двигатся,и где могут возникнуть проблемы. Интересно было бы посмотреть данные в динамики (как меняеся плотижеспособнасть населения в разные временные периуды). 
Если бы обращали больши внимания на данные и анализ то могли бы избежать кризиса 2008 года.

<div style="background: #B0E0E6; padding: 5px; border: 1px solid SteelBlue; border-radius: 5px;">
    <font color='4682B4'><u><b>КОММЕНТАРИЙ</b></u></font>
    <br />
    <font color='4682B4'>
<skj 
     
   Для кредитного отдела банка было проведено иследование в котором разбиралось, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.
Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку. Было выдвинуто четыре гипотезы и установили.
     
                         1 Гипотиза. Есть зависимость между количеством детей и возвратом кредита в срок. 
   
		
* Бездетей выплачивают 13028, должников 1063, что составляет 7.54% (У кого нет детей выплачивают стабильнее, но в будущем они могут появится)
* С 1	ребенкам выплачивают 4364, должников 444, что составляет 9.23 - (У людей с детьми больше финансовых обязательств, кредит платить сложнее)
* С 2 детьми 1858, должников 194, что составляет 9.45 - (Появление второго ребенка не так сильно меняет бюджету и образ жизни как первого.)
* С 3 детьми 303, должников	27, что составляет 8.18 - (Малая выборка, может сильно отклонятся от анализа с большими величинами)
* С 4 детьми 37, должников 4, что составляет 9.76 - (Слишком малая выборка, не является репрезентативной) 
* С 5 детьми 9,	должников 0	- (Слишком малая выборка в расчет можно не брать)

     Таким образом, первая гипотеза подтвердилась лишь отчасти. Этот результат мог оказаться иным, если бы выборка была больше. Да и в течении выпоаты могут появится дети.
  
     
                           2 Гипотиза. Выявили зависимость между семейным положением и возвратом кредита в срок:
    
* Из не женатых / не замужных 9.76%
* Тех кто в разводе 7.06
* У вдовцов / вдов 6.62
* Тех кто в гражданском браке 9.31
* Женатых / замужных 7.56

Вторая гипотеза подтвердилась. Но появилось подозрение, что приволирующим стал возраст, а не семейное положения, для дополнительной гипотизы был сделан срез по возрасту, в нем люди до 35 в среднем чаще невыплачивают задолжность, но кореляция не линейа со своими аномалиями, поэтому возраст можно учитывать, но сильную опору на него делать не стоит.

                         3 Гипотиза. Зависимость между уровнем дохода и возвратом кредита в срок.
* 10.00% Категория E (Доход от 0 до 30000 р.) - (Реже отдают долг, можно предположить, что из за нехватки средств)
* 6.40% Категория D (Доход от 30001 до 50000 р.) - (Самая надежная группа платильщиков)
* 9.29% Категория C (Доход от 50001 до 200000 р.) - (Снова видим увеличения неплатильщиков, возможно из за одобрения слишков высоких кредитов)
* 7.60% Категория B (Доход от 200001 до 1000000 р.) - (снова снижение)
* 8.70% Категория A (Более 1000000 р. - (и снова увеличение). 
     Таким образом, третья гипотеза подтвердилась лишь отчасти. Этот результат мог оказаться иным, если бы не пропуски в данных.
     
                                   4 Гипотиза. Как разные цели кредита влияют на его возврат в срок
* 10.31% операции с автомобилем (Довольно высокий кредит, из за чего сложно его выплачивать, но мативация для выплат вовремя меньше чем у ипотеки)
* 7.82% операции с недвижимостью (Часто залогом является квартира и люди нехотят ее лишится)
* 10.20% получение образования (Сложно учится и полноценно работать одновременно)
* 8.59% проведение свадьбы (Теперь они еще и в категории Женатых / замужных).
     
     Таким образом, четвертая гипотеза подтвердилась. 

**Общие рекомендации.**

* 1. Я бы не советовал опиратся на количество детей. Выборка более трех детей не репрезентативна. А в будущем у клиента могут появится.дети
* 2. Можно использовать семейное положение, в ней прослеживается кореляция, и изменения семейного статуса заемщика скорее всего будет в пользу банка.
* 3. Уровень дохода можно принять к сведению, но не опиратся на него. Из полных данных результат мог бы отличатся.
* 4. Цели кредита рекомендованы к использованию.
* Для лучшего результата нужна более всестороннии иследования, желательно с дополнительными данными.
    </font>
</div>